pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -0,0 +1,187 @@
+                                <NavItem eventKey={"shell"}>
+                                    {_("Shell Script")}
+                                </NavItem>
+                                {this.props.ansible &&
+                                    <NavItem eventKey={"ansible"}>
+                                        {_("Ansible Playbook")}
+                                    </NavItem>
+                                }
+                            </Nav>
+                            <TabContent animation>
+                                <TabPane eventKey={"shell"}>
+                                    <pre>
+                                        {this.props.shell}
+                                    </pre>
+                                </TabPane>
+                                {this.props.ansible &&
+                                    <TabPane eventKey={"ansible"}>
+                                        <pre>
+                                            {this.props.ansible}
+                                        </pre>
+                                    </TabPane>
+                                }
+                            </TabContent>
+                        </React.Fragment>
+                    </TabContainer>
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button bsStyle='default' className='btn' onClick={this.getScript}>
+                        <span className="fa fa-clipboard fa-xs" />
+                        <span>{ _("Copy to clipboard") }</span>
+                    </Button>
+                    <Button bsStyle='default' className='btn-cancel' onClick={this.props.onClose}>
+                        { _("Close") }
+                    </Button>
+                </Modal.Footer>
+            </Modal>
+        );
+    }
+}
+
+ModificationsExportDialog.propTypes = {
+    shell: PropTypes.string.isRequired,
+    ansible: PropTypes.string,
+    show: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+};
+
+/* Display list of modifications in human readable format
+ *
+ * Also show `View automation script` button which opens dialog in which different
+ * scripts are available. With these scripts it is possible to apply the same
+ * configurations to  other machines.
+ *
+ * Pass array `entries` to show human readable messages.
+ * Pass string `shell` and `ansible` with scripts.
+ *
+ */
+export class Modifications extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showDialog: false,
+        };
+    }
+
+    render() {
+        let emptyRow = null;
+        if (this.props.entries === null) {
+            emptyRow = <thead className={"listing-ct-empty"}>
+                <tr className="modification-row">
+                    <td>
+                        <div className="spinner spinner-sm" />
+                        <span>{_("Loading system modification...")}</span>
```

Comment:
_Modification_ is singular here, even though it's plural everywhere else.
Should probably be _modifications_